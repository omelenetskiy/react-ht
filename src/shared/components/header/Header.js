import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import { actionMoviesFetchByQueryString } from '../../store/actions/movies';
import { appTitle } from '../../../config/moviesApp';
import StyledHeader from '../../styled/header';

const StyledLink = styled(Link)`
  color: ${(props) => props.theme.stylishColorLight};
  text-decoration: none;
  border: 2px solid ${(props) => props.theme.stylishColorLight};
  padding: 5px;
  border-radius: 3px;
`;

type Props = {
  location: {
    pathname: string,
  },
};

export const Header = ({ location, dispatch, moviesFetch }: Props) => (
  <StyledHeader>
    <Link to="/404">{appTitle}</Link>
    {location.pathname.includes('/movie') && (
      <StyledLink to="/" onClick={() => moviesFetch()}>
        Search
      </StyledLink>
    )}
  </StyledHeader>
);

const matDispatchToProps = (dispatch) => ({
  moviesFetch: () => dispatch(actionMoviesFetchByQueryString()),
});

export default withRouter(
  connect(
    null,
    matDispatchToProps
  )(Header)
);
