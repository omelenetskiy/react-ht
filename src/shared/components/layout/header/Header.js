import React from 'react';
import styled from 'styled-components';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';

import { appTitle } from '../../../../config/moviesApp';
import StyledHeader from '../../styled/header';

const StyledLink = styled(Link)`
  color: ${(props) => props.theme.stylishColorLight};
  text-decoration: none;
  border: 2px solid ${(props) => props.theme.stylishColorLight};
  padding: 5px;
  border-radius: 3px;
`;

type Props = {
  location: {
    pathname: string,
  },
};

export const Header = ({ location }: Props) => (
  <StyledHeader>
    <Link to="/">{appTitle}</Link>
    {location.pathname !== '/' && <StyledLink to="/">Search</StyledLink>}
  </StyledHeader>
);

export default withRouter(Header);
