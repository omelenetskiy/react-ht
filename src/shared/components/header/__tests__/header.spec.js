import React from 'react';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Header, StyledLink } from '../Header';
import 'jest-enzyme';
import 'jest-styled-components';

const initialState = {};
const mockStore = configureStore();
let store;

jest.mock('../../../../config/moviesApp', () => ({
  appTitle: 'appTitle',
}));

const theme = {
  stylishColorLight: 'stylishColorLight',
};

const moviesFetch = jest.fn();

const props = {
  moviesFetch,
  dispatch: jest.fn(),
};

describe('Header', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });
  it('should render correctly', () => {
    const location = {
      pathname: '/',
    };
    const wrapper = (
      <Provider store={store}>
        <Router>
          <Header location={location} />
        </Router>
      </Provider>
    );
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const location = {
      pathname: '/movie',
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <Header {...props} location={location} />
        </Router>
      </Provider>
    );
    const linkWrapper = shallow(
      <Router>
        <StyledLink to="/" theme={theme} {...props} />
      </Router>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(linkWrapper)).toMatchSnapshot();
  });
});
