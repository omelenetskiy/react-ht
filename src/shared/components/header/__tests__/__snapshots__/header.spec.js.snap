// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
.c0 {
  font-weight: bold;
  min-height: 30px;
  padding: 10px 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 2px solid;
  padding: 5px;
  border-radius: 3px;
}

<header
  className="c0"
>
  <a
    href="/"
    onClick={[Function]}
  >
    appTitle
  </a>
  <a
    className="c1"
    href="/"
    onClick={[Function]}
  >
    Search
  </a>
</header>
`;
